src/formatDate.js

```diff
@@ -8,7 +8,35 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const delimiters = ['-', '/', '.'];
+  const inputDelimiter = delimiters.find((d) => date.includes(d));
+  const dateParts = date.split(inputDelimiter);
+
+  const dateObj = {};
+
+  fromFormat.forEach((part, index) => {
+    dateObj[part] = dateParts[index];
+  });
```

Comment:
There is no validation to ensure that `fromFormat` and `dateParts` have the same length. This could lead to undefined behavior if they do not match.